### ISSUE TITLE
Randomly gaining/not gaining skill points.

### ISSUE THREAD

**Issue by Fapboard** [darklordabc/Legends-of-Dota-Redux#1318]
Frequently ingame a hero stops gaining skill points on level up per usual after the 20 mark and starts doing so again on 25.
Thats 4 skill points which goes into the void, usually frustrates me when my build suddenly becomes incomplete and shouldnt happen in the first place.

Was haphazzardly fixed by raising the level cap before, now it was placed back on 28 again because of a recent and presumable unfixed XP glitch (which causes heroes to ding 100 after a single kill when lv.30 ish).

My suggestion is to just simply giving any player reaching level 28 4 free skill points thus ensuring if the bug occurs.

I've also noticed that you inexplicably gain skill points during a time window when an ally dies.
And on separate ocations gaining skill points whenever your team score kills.

Seems like a switch behind the scenes (up to 3) that goes on/off for unknown reasons/triggers which has certain effects on how players wether gain skill points or not.

Probably a glitch in the code that causes it to skip a line or intervine with another integer that unintentionally triggers this pointless feature (which could actually be a fun game mode/mod for dota map if pinpointed and extracted).

Until then I sugesst applying my suggestion mentioned above.

**Comment by Darkmega18** [comment]
it actually doesn't occur anymore, the instant level 100 glitch, since haven't had it happen in the past week or so. so I always set my games to level 100 and it's fine. although the levels do really shoot straight up into the stratosphere regardless once you get past level 25. especially when a lot of kills are included.

edit: [this was a derp response. didn't read the topic or didn't see it properly cause of thinking of something, I dunno, but I don't like deleting posts cause it leaves holes. But there was mention of a ding to level 100 glitch in your post. which could've been a reason when I was reading it.]

**Comment by Fapboard** [comment]
Yes it does. Happens to me everyday.

That is the reason I made this thread.

Please remove your comment/response from this thread.
 - Thanks in advance.

**Comment by Darkmega18** [comment]
we're either playing a different game then or have different versions or play with different settings that upset the chance of it happening then. cause like I said, haven't had it for weeks. otherwise I wouldn'tve been able to have as much fun doing my planet cracker singleplayer build. mind you, for benefit of the doubt, I could've got lucky just cause of the sheer quantity of XP flying around when I was doing these.

we need a third opinion. :V

**Comment by Fapboard** [comment]
To cover a basic potentiall missunderstanding:
I play the first map: All_Allowed primarily.
And you seem to play Dota map.

These two maps are diffrent in up to date with things, this particular problem concerns and happens frequently on All_Allowed map.

In the meantime, Dota map has it's own problems, like the loading issues by featuring the old system via the client for basic functions while the ingame hero picker bypassess potential misshappenings from the clients-side by using a mod function for drafting and picking abilities, quite genius btw and might be nescessary to be applied on the dota map as well.

And to throw the punchline on this particular topic is that both these maps has diffrent progression on their development, with Dota map being behind in updates.
 - Actually main reason I dont host Dota map, since many players fails to load by default from the game engine (and in turn deserves it's on topic/discussion on this forum).

Now i'll say this once again, your response does not apply nor help the the case here and ought to be removed from the thread.

And since I see darklord had placed a thumbsdown on my previous response, I urge you to respond on this topic of how you will proceed/adress this issue.

**Comment by Fapboard** [comment]
![image](https://user-images.githubusercontent.com/30776198/34228848-796541b6-e5d2-11e7-857b-64ce31d25d2d.png)

Case of missing skill points which was not awarded after level up beyond 20.

Total of:
Rearm redux: 3p
Conc Shot: 4p
Conc Shot 2: 3p
Wrath: 2p
Echo: 3p
Talents: 3p
Basic maths -> 18 skill points distributed in total at level 22, 4 is missing.

What gives.

**Comment by Fapboard** [comment]
After previous update, the occurance still remains.

![image](https://user-images.githubusercontent.com/30776198/34343532-5b670938-e9d3-11e7-802d-d6737025c303.png)

As previously mentioned, the issue of lacking skill points has been mitigated by raising the level cap once again, however the bug still remains where you stop aquiring skill points on level up after the 20 mark.

**Comment by Fapboard** [comment]
Another day another occurance.

![image](https://user-images.githubusercontent.com/30776198/34387232-3b52b1aa-eb2d-11e7-9d14-2f281f37fc30.png)

As a sidenote: I have not seen the 100 level glitch at all since latest update.

Ontopic, stopped aquiring skill points past level 20 on the default map.

**Comment by darklordabc** [comment]
Any idea what is causing this?

**Comment by Fapboard** [comment]
In response to Darklordabc's reply:
"Any idea what is causing this?"

None what'so ever actually.

All I have gathered from this is a pattern.
Pattern being the following: 
A switch is triggered in the background at 20 -> do not grant skill point on lvl up.
A switch is triggered in the backround at 25 -> grant skill points on level up.

I've also observed that another is switch is triggered ocationally:
On ally death -> Grant a skill point.
This switch seems to switch on when enemy team is leading with kill advantage (roughly 15 kills) during early to mid game. This is fairly uncertain and unconfirmed as of yet and could use some further investigation. Still being an occurence is certain however.

My best guess is that some line of code (possibly a left-over one) that is triggered by event of ding 20.
Possibly by a previous developer, or a game modifier is interacting when it shouldn't or a pure accident/slip of code written by accident or is missbehaving.

A shot in the dark is what the dota map and default map has in diffrence is the loader, ingame builder at start and might be helpful to compare these 2 maps(/mods or what you call it) diffrence in code.

It is very clear however that there is a cause which is unknown.
And if youre in the blue on this one, i'd simply suggest starting looking into the code of potential culprits (personally i'd recomend some of the suspects i've mentioned above.

As a player I specificly dont know what's causing it.

TL:DR:
No, your guess is as good as mine (or better).

**Comment by darklordabc** [comment]
Could it be that it stuffs up when there is a disconnected player?

**Comment by Fapboard** [comment]
No.

**Comment by Fapboard** [comment]
With latest fix applied by dankbud, that checks each level if an individual has correct amount of skill points for his level:

- It works and this issue is resolved.

Thread is to be closed.
